cogs/upgrade.py

```diff
@@ -0,0 +1,157 @@
+from core.logger import get_discord_child_logger
+from mongo.study_subject_relation import StudySubjectRelations, StudySubjectRelation
+from mongo.subjects_or_groups import SubjectsOrGroups, SubjectOrGroup
+
+logger = get_discord_child_logger("upgrade")
+
+
+class Upgrade(Cog):
+    """
+    Provides the upgrade command
+    """
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @command(help="Performs the upgrade")
+    @is_owner()
+    async def upgrade(self, ctx: Context):
+        logger.info("Started doing upgrade")
+
+        guild: Guild = ctx.guild
+        members: list[Member] = guild.members
+
+        study_groups_db = SubjectsOrGroups(self.bot, SubjectsOrGroupsEnum.GROUP)
+        study_groups_documents: list[SubjectOrGroup] = await study_groups_db.find({})
+
+        subject_db = SubjectsOrGroups(self.bot, SubjectsOrGroupsEnum.SUBJECT)
+        subjects_documents: list[SubjectOrGroup] = await subject_db.find({})
+        subject_roles: list[Role] = [document.role for document in subjects_documents]
+
+        links_db = StudySubjectRelations(self.bot)
+        links_documents: list[StudySubjectRelation] = await links_db.find({})
+
+        match = re.compile(r"([a-z]+)([0-9]+)", re.I)
+
+        # remove every subject from everyone:
+        logger.info("Starting to remove subjects from everyone")
+        for member in members:
+            logger.info(f"Checking {member.display_name}")
+            roles_to_remove = [role for role in subject_roles]
+            await member.remove_roles(*roles_to_remove, reason="upgrade")
+        logger.info("Finished removing subjects from everyone")
+
+        # recreate subject channels:
+        logger.info("Starting to recreate the subject text channels")
+        for document in subjects_documents:
+            channel: TextChannel = document.chat
+            name = channel.name
+            category = channel.category
+            permissions = channel.overwrites
+            new_channel = await guild.create_text_channel(name=name,
+                                                          category=category,
+                                                          overwrites=permissions,
+                                                          nsfw=False,
+                                                          reason="upgrade")
+            try:
+                await channel.delete(reason="upgrade")
+            except NotFound:
+                pass
+
+            logger.info(f"Recreated {name} channel")
+
+            new_document = {
+                DBKeyWrapperEnum.CHAT.value: new_channel.id,
+                DBKeyWrapperEnum.ROLE.value: document.role_id
+            }
+            await subject_db.update_one(document.document, new_document)
+        logger.info("Finished to recreate the subject text channels")
+
+        # update the documents with the new ones
+        subjects_documents: list[SubjectOrGroup] = await subject_db.find({})
```

Comment:
This variable is assigned a value but never used afterwards.
Remove or use the variable.